pingcap/tidb-operator
# update sync order for the components

### What problem does this PR solve?

Fix https://github.com/pingcap/tidb-operator/issues/3853
### What is changed and how does it work?

Discussed with the dev team of different components, the order to sync the components should be:
CDC, PD, TiKV, TiFlash/Pump, TiDB
### Code changes


### Tests


  - Deploy TidbCluster with CDC, PD, TiKV, TiFlash, Pump, TiDB, waiting for Pods ready, upgrade the tc, check that the upgrade order is CDC, TiFlash, PD, TiKV, Pump, TiDB
  - Deploy TidbCluster with CDC, PD, TiKV, TiFlash, Pump, TiDB and enable TLS, waiting for Pods ready, upgrade the tc, check that the upgrade order is CDC, TiFlash, PD, TiKV, Pump, TiDB
  - Deploy TidbCluster with PD, TiKV, TiFlash, TiDB, waiting for Pods ready, upgrade the tc, check that the upgrade order is TiFlash, PD, TiKV, TiDB
  - Deploy TidbCluster with PD, TiKV, TiFlash, TiDB and enable TLS, waiting for Pods ready, upgrade the tc, check that the upgrade order is TiFlash, PD, TiKV, TiDB

### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
NONE
```
